### ISSUE TITLE
Should segment() take care of leading slash?

### ISSUE THREAD

**Issue by orlando** [medialize/URI.js#236]
I'm using URI.js to append a cdn host to URLs, the problem is that some assets have a leading slash and other don't.

Should `segment()` method take care of this?

## Example

```
URI('https://google.com').segment('font.ttf').toString() //-> https://google.com/font.ttf
URI('https://google.com').segment('/font.ttf').toString() //-> https://google.com//font.ttf
```
Result of second example should be like the first one.

Thanks!

**Comment by rodneyrehm** [comment]
is it possible you're looking for [`.absoluteTo()`](http://medialize.github.io/URI.js/docs.html#absoluteto) instead?

```js
URI('font.ttf').absoluteTo('https://google.com/').toString()
// https://google.com/font.ttf
URI('/font.ttf').absoluteTo('https://google.com/').toString()
// https://google.com/font.ttf
URI('font.ttf').absoluteTo('https://google.com/somewhere/').toString()
// https://google.com/somewhere/font.ttf
URI('/font.ttf').absoluteTo('https://google.com/somewhere/').toString()
// https://google.com/font.ttf
```

But to answer your question: the [`.segment()`](http://medialize.github.io/URI.js/docs.html#accessors-segment) function is nothing but a helper to split and join paths. There also is [`.segmentCoded()`](http://medialize.github.io/URI.js/docs.html#accessors-segmentCoded) to take care of en/decoding segments. There is no logic along the lines of "is the value to set a segment, or really an entire path". I'm not sure this should be added, as you may very well replace a given segment by `foo/bar`, which is valid. 

Removing leading and trailing slashes would be a convenience thing. Feel free to submit a PR for that :)

**Comment by orlando** [comment]
@rodneyrehm `abosoluteTo()` works for this use case.

if you think `segment()` should do this, I can implement it. Do you have any suggestion about the API, or instead sanitize leading/trailing slashes by default?

**Comment by rodneyrehm** [comment]
> if you think segment() should do this, I can implement it. Do you have any suggestion about the API, or instead sanitize leading/trailing slashes by default?

leading and trailing `/` make no sense in a segment, so'd trim them by default.

**Comment by orlando** [comment]
Perfect, I'll work on this over the weekend.

Thanks!

**Comment by rodneyrehm** [comment]
any progress on this, @orlando?

**Comment by orlando** [comment]
@rodneyrehm actually I made the fix, but totally forgot about the PR. Thanks for the heads up, I'll make the PR between tonight and tomorrow.

**Comment by rodneyrehm** [comment]
any progress on the PR, @orlando?

**Comment by orlando** [comment]
@rodneyrehm PR up

**Comment by rodneyrehm** [comment]
released in [v1.17.0](https://github.com/medialize/URI.js/releases/tag/v1.17.0)
